controllers/extendeddaemonset/controller.go

```diff
@@ -510,6 +526,13 @@ func (r *Reconciler) cleanupReplicaSet(logger logr.Logger, rsList *datadoghqv1al
 	return utilserrors.NewAggregate(errs)
 }
 
+func clearCanaryAnnotations(eds *datadoghqv1alpha1.ExtendedDaemonSet) {
+	delete(eds.Annotations, datadoghqv1alpha1.ExtendedDaemonSetCanaryPausedAnnotationKey)
```

Comment:
maybe we can. but I'm not sure it will improve the readability or decrease the number of line since we will need to create a slice of annotation's keys.